ArtOfIllusion/src/artofillusion/image/ExternalImage.java

```diff
@@ -0,0 +1,479 @@
+/* Copyright (C) 2017 by Petri Ihalainen
+
+   This program is free software; you can redistribute it and/or modify it under the
+   terms of the GNU General Public License as published by the Free Software
+   Foundation; either version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY 
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A 
+   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */
+
+package artofillusion.image;
+
+import artofillusion.*;
+import artofillusion.math.*;
+import buoy.widget.BStandardDialog;
+import java.awt.*;
+import java.awt.image.*;
+import java.io.*;
+import java.util.Date;
+import javax.imageio.*;
+
+public class ExternalImage extends ImageMap
+{
+  private File externalFile;
+  private String loadingError, imageType;
+  private ImageMap imageMap;
```

Comment:
How about this:

- Keep the `ImageMap` as the super type, that textures communicate with.
- Have two direct sub types: `ExternalImage` and `MappedImage` (or such).
- The actual maped image types (HDR, SVG, MIP...) would be sub types of `MappedImage`.
- `ExternalImage` would wrap a `MappedImage`, _not_ another `ImageMap`.

That would make a clear separation between the classes and get rid of any overlapping and it seems a relatively simple task to do.

Earlier Peter said something about the ExternalImage _implementing_ ImageMap but after a short test I went back to extending (don't remember the details, but it seemed simpler that way). On general level I was trying follow the example of `ObjectWrapper`, where both the wrappers and the objects are extending Object3D. But...

Currently ImageMap has (at least) the method, that chooses which image type is going to me mapped, that ExternalImage does not use.t. On the other hand, if `ExternalImage` were to be merged with `ImageMap`, then all the functions, that are needed to handle the external link _and what ever else comes with that_, would be unnecessary to the internally mapped images and would actully need to be turned off or change their function in that case.

How does that sound to you?